Core/Core/ActAsUser/ActAsUserViewController.swift

```diff
@@ -18,121 +18,142 @@
-        userIDTextField.addTarget(self, action: #selector(enterPressed), for: .editingDidEndOnExit)
+        domainTextField.addTarget(self, action: #selector(actAsUserPressed), for: .editingDidEndOnExit)
+        userIDTextField.addTarget(self, action: #selector(actAsUserPressed), for: .editingDidEndOnExit)
 
-        redPanda.transform = redPanda.transform.rotated(by: .pi / -6).translatedBy(x: 0, y: 0)
-        actAsUserButton.layer.cornerRadius = 5
-        animatePanda()
+        NotificationCenter.default.addObserver(self, selector: #selector(keyboardWillChangeFrame(_:)), name: UIResponder.keyboardWillChangeFrameNotification, object: nil)
     }
 
     public override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
+        keyboard = KeyboardTransitioning(view: view, space: keyboardSpace)
         navigationController?.navigationBar.useModalStyle()
+        animatePanda()
     }
 
     func animatePanda() {
-        DispatchQueue.main.asyncAfter(deadline: .now() + .seconds(2)) { [weak self] in
-            self?.redPandaTopConstraint.constant -= 30
-            self?.redPandaTrailingConstraint.constant -= 82
-            UIView.animate(withDuration: 1.5, animations: { [weak self] in
-                guard let redPanda = self?.redPanda else {
-                    return
-                }
+        let animation = CAKeyframeAnimation(keyPath: "transform")
+        let easeInEaseOut = CAMediaTimingFunction(name: .easeInEaseOut)
+        let cover = CATransform3DIdentity
+        let aside = CATransform3DMakeAffineTransform(redPanda.transform.translatedBy(x: -82, y: 30).rotated(by: .pi / -6))
+        animation.timingFunctions = [ easeInEaseOut, easeInEaseOut, easeInEaseOut, easeInEaseOut ]
+        animation.duration = 10
+        animation.keyTimes = [ 0, 0.2, 0.35, 0.85, 1 ]
+        animation.values = [ aside, aside, cover, cover, aside ]
+        animation.repeatCount = .infinity
+        redPanda.layer.add(animation, forKey: "masquerade")
+    }
 
-                redPanda.transform = redPanda.transform
-                    .rotated(by: .pi / 6)
-                self?.view.layoutIfNeeded()
-                }, completion: { [weak self] _ in
-                    self?.redPandaTopConstraint.constant += 30
-                    self?.redPandaTrailingConstraint.constant += 82
-                    UIView.animate(withDuration: 1.5, delay: 5, animations: { [weak self] in
-                        guard let redPanda = self?.redPanda else {
-                            return
-                        }
-                        redPanda.transform = redPanda.transform
-                            .rotated(by: .pi / -6)
-                        self?.view.layoutIfNeeded()
-                    }, completion: { [weak self] _ in
-                        self?.animatePanda()
-                    })
-            })
-        }
+    @objc func keyboardWillChangeFrame(_ notification: Notification) {
```

Comment:
my tests aren't yet covering this function, which tries to redo what Matt wanted from the UITableViewController originally. I can add tests to cover this before we merge.